File: WebUI/client-app/src/app/models/reservation.ts
export interface IReservationsVm {
  reservations: IReservation[];
  reservationsCount: number;
}

export interface IReservation {
  id: number;
  startTime: string;
  endTime: string;
  date: Date;
  createdAt: Date;
  price: number;
  hairSalonId: number;
  hairSalon: IHairSalon;
  userId: string;
  user: IUser;
  statusId: number;
  status: IReservationStatus;
}

interface IHairSalon {
  id: number;
  name: string;
}

interface IUser {
  id: string;
  username: string;
}

interface IReservationStatus {
  id: number;
  status: string;
}

export interface IFreeTermsVm {
  hairSalonId: number;
  freeTerms: IFreeTerm[];
  freeTermsCount: number;
}

export interface IFreeTerm {
  date: Date;
  startTime: string;
  price: number;
}

export interface ITermsVm {
  hairSalonId: number;
  termsByDate: ITermByDate[];
  termsCount: number;
}

export interface ITermByDate {
  date: Date;
  terms: ITerm[];
}

export interface ITerm {
  startTime: string;
  price: number;
  status: string;
  isExpired: boolean;
}

File: Domain/Entities/Reservation.cs
using System;

namespace Domain.Entities
{
    public class Reservation
    {
        public int Id { get; set; }
        public int HairSalonId { get; set; }
        public string UserId { get; set; }
        public TimeSpan StartTime { get; set; }
        public TimeSpan EndTime { get; set; }
        public DateTime Date { get; set; }
        public DateTime CreatedAt { get; set; }
        public int Price { get; set; }
        public int StatusId { get; set; }

        public virtual HairSalon HairSalon { get; set; }
        public virtual AppUser User { get; set; }
        public virtual ReservationStatus Status { get; set; }
    }
}